pkg/operator/controller/names.go

```diff
@@ -120,3 +134,74 @@ func NodeResolverDaemonSetPodSelector() *metav1.LabelSelector {
 		},
 	}
 }
+
+// CABundleCMVolAndVolMount takes a CA bundle ConfigMap name and a TLS server name, and returns
+// the ConfigMap Volume and VolumeMount to be used for the DaemonSet.
+func CABundleCMVolAndVolMount(caBundleName string, serverName string, caBundleRevisionMap map[string]string) (*corev1.Volume, *corev1.VolumeMount) {
+	caBundleConfigmapName := CABundleConfigMapName(caBundleName)
+	caBundleVolumeName := caBundleConfigmapName.Name
+	caBundleFilename := "ca-bundle.crt"
+	caBundleVolume := corev1.Volume{
+		Name: caBundleVolumeName,
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: caBundleVolumeName,
+				},
+				Items: []corev1.KeyToPath{
+					{
+						Key:  caBundleFilename,
+						Path: caBundleFilename,
+					},
+				},
+			},
+		},
+	}
+	caBundleVolumeMountPath := fmt.Sprintf("/etc/pki/%s-%s", serverName, caBundleRevisionMap[caBundleName])
+	caBundleVolumeMount := corev1.VolumeMount{
+		Name:      caBundleVolumeName,
+		MountPath: caBundleVolumeMountPath,
+		ReadOnly:  true,
+	}
+	return &caBundleVolume, &caBundleVolumeMount
+}
+
+// CABundleRevisionMap generates a map of ca bundle configmaps with their resource versions
+// to be used in Corefile as the path of ca bundles. Resource version is appended to enable automatic
+// reload of Corefile when there is a change in the source ca bundle configmap (e.g. cert rotation).
+func CABundleRevisionMap(client client.Client, dns *operatorv1.DNS) (map[string]string, error) {
```

Comment:
Same with this function, I think it fit better in pkg/operator/controller/controller_dns_configmap.go